cardano-db/src/Cardano/Db/Migration/Extra/CosnumedTxOut/Queries.hs

```diff
@@ -170,10 +146,220 @@ _validateMigration trce = do
                 ]
           pure False
 
+--------------------------------------------------------------------------------------------------
+-- Inserts
+--------------------------------------------------------------------------------------------------
+insertTxOutExtra :: (MonadBaseControl IO m, MonadIO m) => TxOut -> ReaderT SqlBackend m TxOutId
+insertTxOutExtra = insertUnchecked "TxOutExtra"
+
+insertManyTxOutExtra :: (MonadBaseControl IO m, MonadIO m) => [TxOut] -> ReaderT SqlBackend m [TxOutId]
+insertManyTxOutExtra = insertMany' "TxOut"
+
+--------------------------------------------------------------------------------------------------
+-- Updates
+--------------------------------------------------------------------------------------------------
+updateTxOutConsumedByTxInId :: MonadIO m => TxOutId -> TxInId -> ReaderT SqlBackend m ()
+updateTxOutConsumedByTxInId txOutId txInId =
+  update txOutId [TxOutConsumedByTxInId =. Just txInId]
+
+-- | This requires an index at TxOutConsumedByTxInId.
+getTxOutConsumedAfter :: MonadIO m => TxInId -> ReaderT SqlBackend m [TxOutId]
+getTxOutConsumedAfter txInId = do
+  res <- select $ do
+    txOut <- from $ table @TxOut
+    where_ (txOut ^. TxOutConsumedByTxInId >=. just (val txInId))
+    pure $ txOut ^. persistIdField
+  pure $ unValue <$> res
+
+-- | This requires an index at TxOutConsumedByTxInId.
+setNullTxOutConsumedAfterTxInId :: MonadIO m => TxOutId -> ReaderT SqlBackend m ()
+setNullTxOutConsumedAfterTxInId txOutId = do
+  update txOutId [TxOutConsumedByTxInId =. Nothing]
+
+migrateTxOut ::
+  ( MonadBaseControl IO m
+  , MonadIO m
+  ) =>
+  Maybe (Trace IO Text) ->
+  ReaderT SqlBackend m ()
+migrateTxOut mTrace = do
+  _ <- createConsumedTxOut
+  migrateNextPage 0
+  where
+    migrateNextPage :: MonadIO m => Word64 -> ReaderT SqlBackend m ()
+    migrateNextPage offst = do
+      whenJust mTrace $ \trce ->
+        liftIO $ logInfo trce $ "Handling input offset " <> textShow offst
+      page <- getInputPage offst pageSize
+      mapM_ migratePair page
+      when (fromIntegral (length page) == pageSize) $
+        migrateNextPage $!
+          offst
+            + pageSize
+
+migratePair :: MonadIO m => (TxInId, TxId, Word64) -> ReaderT SqlBackend m ()
+migratePair (txInId, txId, index) =
+  updateTxOutConsumedByTxInIdUnique txId index txInId
+
+--------------------------------------------------------------------------------------------------
+-- Delete + Update
+--------------------------------------------------------------------------------------------------
+deleteAndUpdateConsumedTxOut ::
+  forall m.
+  (MonadIO m, MonadBaseControl IO m) =>
+  Trace IO Text ->
+  Word64 ->
+  ReaderT SqlBackend m ()
+deleteAndUpdateConsumedTxOut trce blockNoDiff = do
+  maxTxInId <- findMaxTxInId blockNoDiff
+  case maxTxInId of
+    Left errMsg -> do
+      liftIO $ logInfo trce $ "No tx_out was deleted: " <> errMsg
+      migrateNextPage Nothing False 0
```

Comment:
```suggestion
      migrateTxOut (Just trce)
```
This may also work. I saw you initially did it [here](https://github.com/input-output-hk/cardano-db-sync/compare/6836212277ea853ca00f29116ec141e5a99c8380..d034f391b42423c4426912590530597d6fb94b8a#diff-f2320ab911286735a4cc1c93a700fd7fe3f14fe3d2400201823b93fa916190c0L237), but then you changed it back

Reply:
I made those changes following your previous comments https://github.com/input-output-hk/cardano-db-sync/pull/1519#discussion_r1340198363

Reply:
The comment was one day before this commit.

Reply:
One of your comments mentioned:
> It could take a `Maybe` and only update if it's `Nothing`

> this case should still call `migrateNextPage` and update for all `tx_in` entries.

So I made the changes as advised but you're saying put it back to just doing `migrateTxOut (Just trce)` rather than pass the `Maybe`?

Reply:
A misunderstanding then. I commended when the code looked like 
```suggestion
      createConsumedTxOut
      liftIO $ logInfo trce "Created ConsumedTxOut as there were no blocks present"
```
and hadn't seen https://github.com/input-output-hk/cardano-db-sync/pull/1519/commits/d649e18ec0f37e20c1f027eb68bfe9c8c3c1ff75, which actually looks like a good idea, unless I'm missing something.

Reply:
So either way, the current or the above should be fine.